src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Filterable.java

```diff
@@ -1,10 +1,16 @@
 package io.stargate.sgv2.jsonapi.api.model.command;
 
 import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.FilterClause;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.FilterSpec;
 
 /*
  * All the commands that needs FilterClause will have to implement this.
  */
 public interface Filterable {
-  FilterClause filterClause();
+  FilterSpec filterSpec();
```

Comment:
To separate out low-level JSON decoding and higher-level deserialization/marshalling, new "FilterSpec" (which only has raw JSON as `JsonNode`) added: `FilterClause` remains the fully-processed definition to use, produced from `FilterSpec`